### PR TITLE
[embedded] Avoid use of a metatype in array implementation for embedded Swift

### PR DESCRIPTION
**Pull request by kubamracek**

### CODE REVIEW

**Comment by kubamracek** [comment]
@swift-ci please test

**Comment by kubamracek** [comment]
@swift-ci please test macOS platform

**Review by phausler — APPROVED** [review]

File: stdlib/public/core/ArrayShared.swift [hunk 1]
```diff
@@ -41,9 +41,14 @@ func _allocateUninitializedArray<Element>(_  builtinCount: Builtin.Word)
   if count > 0 {
     // Doing the actual buffer allocation outside of the array.uninitialized
     // semantics function enables stack propagation of the buffer.
+    let storageType: _ContiguousArrayStorage<Element>.Type
+    #if !$Embedded
```
Comment (slavapestov) [inline stdlib/public/core/ArrayShared.swift:45]: Would it work to sink this down to getContiguousArrayStorageType()?

Comment (kubamracek) [reply stdlib/public/core/ArrayShared.swift:45]: I tried that originally, but the function call that returns a metatype is a problem in embedded Swift. I think we could improve the MandatoryPerfOptimizations to handle that case, but I think I'd like leave that option as a possible follow-up improvement.

**Pull request merged**